sympy/functions/special/gamma_functions.py

```diff
@@ -993,7 +993,10 @@ def digamma(x):
     .. [2] http://mathworld.wolfram.com/DigammaFunction.html
     .. [3] http://functions.wolfram.com/GammaBetaErf/PolyGamma2/
     """
-    return polygamma(0, x)
+    def __new__(cls, x, evaluate=True):
+        if evaluate:
+            return polygamma(0, x, evaluate=evaluate)
```

Comment:
Is it possible to evaluate digamma into polygamma later?

Reply:
@oscargus I didn't understand. The idea was to keep it polygamma when evaluate=True and digamma when evaluate=False.

Reply:
Calling `digamma(x, evaluate=False).doit()` will restore to `polygamma` since it reconstructs the node.
My idea here is that it is better to have some options to bypass automatic evaluation.